pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/realtime/SegmentCompletionManager.java

```diff
@@ -996,12 +995,12 @@ private int numReplicasToLookFor() {
       // In case of splitCommit, the segment is uploaded to a unique file name indicated by segmentLocation,
       // so we need to move the segment file to its permanent location first before committing the metadata.
       if (isSplitCommit) {
-        if (!_segmentManager.commitSegmentFile(_segmentName.getTableName(), segmentLocation, _segmentName.getSegmentName())) {
+        if (!_segmentManager.commitSegmentFile(_segmentName.getTableName(), reqParams.getSegmentLocation(),
```

Comment:
Use CommittingSegmentDescriptor when calling this API. Keep the usage of ReqParams local to this file